src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/merge_transpose_reorder.cpp

```diff
@@ -0,0 +1,176 @@
+                                                                              ov::Strides{1, 1},
+                                                                              ov::CoordinateDiff{1, 1},
+                                                                              ov::CoordinateDiff{1, 1},
+                                                                              ov::Strides{1, 1});
+        // It's necessary to force acdb layout to be sure that the reorder, which changes dims order, will be inserted
+        // (by default acdb layout is chosen only on >= AVX512 platforms)
+        const auto conv = conv_with_bias->get_input_node_shared_ptr(0);
+        const auto acdb_format = CPUTestUtils::cpu_memory_format_t::acdb;
+        conv->get_rt_info() = makeCPUInfo({acdb_format}, {acdb_format}, {});
+
+        const auto dim_h = ov::op::util::node_to_get_shape_value_of_indices_from_shape_source(param, {2}, {}, shapeof_subgraph_prc);
+        const auto dim_w = ov::op::util::node_to_get_shape_value_of_indices_from_shape_source(param, {3}, {}, shapeof_subgraph_prc);
+        const auto fused_spatial_dims = ov::op::util::make_try_fold<ov::opset10::Multiply>(dim_h, dim_w);
+        const auto reshape_const_3 = ov::op::util::make_try_fold<ov::opset10::Concat>(ov::OutputVector{unchangable_dims, fused_spatial_dims}, 0);
+        const auto reshape_3 = std::make_shared<ov::opset10::Reshape>(conv_with_bias, reshape_const_3, true);
+        const auto transpose_const_3 = ov::opset10::Constant::create(shapeof_subgraph_prc, {3}, {0, 2, 1});
+        const auto transpose_3 = std::make_shared<ov::opset10::Transpose>(reshape_3, transpose_const_3);
+
+        const auto fc_weights_2 = ngraph::builder::makeConstant(precision, ov::Shape{channels, fc_out_channels}, std::vector<float>{}, true);
+        const auto fc_2 = make_layer_with_bias<ov::opset10::MatMul>(transpose_3, fc_weights_2, false, true);
+        function = std::make_shared<ov::Model>(fc_2, ov::ParameterVector{param}, "MergeTransposeReorderModel");
+    }
+
+    void validate_exec_graph() {
+        const size_t original_reshape_transpose_count = 3;
+        const size_t non_optimized_reshape_transpose_count = original_reshape_transpose_count - m_optimized_reorders_count;
+        CheckNumberOfNodesWithType(compiledModel, "Transpose", non_optimized_reshape_transpose_count);
+        CheckNumberOfNodesWithType(compiledModel, "Reshape", non_optimized_reshape_transpose_count);
+        size_t fake_reorder_count = 0;
+        for (const auto& node : compiledModel.get_runtime_model()->get_ops()) {
+            const auto& rtInfo = node->get_rt_info();
+            auto it = rtInfo.find(ov::exec_model_info::LAYER_TYPE);
+            IE_ASSERT(rtInfo.end() != it);
+            if (it->second.as<std::string>() == "Reorder" && node->get_friendly_name().find("_fake") != std::string::npos) {
+                fake_reorder_count++;
+            }
```

Comment:
Such type of a check is not really reliable as the name may be changed by some reason. Could you please elaborate on the reason why we need to count only "fake" reorders? Can we count all the reorders?

Reply:
Yes, I agree: the checks are done in a more reliable way now